pandas/_libs/algos.pyx

```diff
@@ -1179,7 +1180,7 @@ def diff_2d(diff_t[:, :] arr,
             Py_ssize_t periods, int axis):
     cdef:
         Py_ssize_t i, j, sx, sy, start, stop
-        bint f_contig = arr.flags.f_contiguous
+        bint f_contig = PyBuffer_IsContiguous(arr, 'F')
```

Comment:
neat.  probably not worth the effort on our end, but itd be nice if cython automatically made this optimization

Reply:
tbh didn't look too much into the subsequent code; think could be cleaned up?